src/Photons.ts

```diff
@@ -138,63 +133,66 @@ export default class Photons {
 
   /**
    * Create a propagator, given this object dimX and dimY.
+   * Propagator "moves" photons.
    * @param yDirMeansDown For true, direction 'v' increments dimY.
+   * @todo is yDirMeansDown really necessary? It is only present in this file.
    *
    * @return An operator, with dimensions [dimX, dimY, {@link Dimension.direction()}].
    */
   createPhotonPropagator(yDirMeansDown = true): Operator {
     const dir = Dimension.direction()
-    const dimX = this.dimX
-    const dimY = this.dimY
     const s = yDirMeansDown ? 1 : -1
 
     return Operator.add([
-      Operator.outer([Operator.shift(dimX, +1), Operator.identity([dimY]), Operator.indicator([dir], ['>'])]),
-      Operator.outer([Operator.shift(dimX, -1), Operator.identity([dimY]), Operator.indicator([dir], ['<'])]),
-      Operator.outer([Operator.identity([dimX]), Operator.shift(dimY, +s), Operator.indicator([dir], ['v'])]),
-      Operator.outer([Operator.identity([dimX]), Operator.shift(dimY, -s), Operator.indicator([dir], ['^'])]),
+      Operator.outer([Operator.shift(this.dimX, +1), Operator.identity([this.dimY]), Operator.indicator([dir], ['>'])]),
+      Operator.outer([Operator.shift(this.dimX, -1), Operator.identity([this.dimY]), Operator.indicator([dir], ['<'])]),
+      Operator.outer([Operator.identity([this.dimX]), Operator.shift(this.dimY, +s), Operator.indicator([dir], ['v'])]),
+      Operator.outer([Operator.identity([this.dimX]), Operator.shift(this.dimY, -s), Operator.indicator([dir], ['^'])]),
     ])
   }
 
   /**
    * Propagate all particles, using {@link createPhotonPropagator}.
    * @param yDirMeansDown or true, direction 'v' increments dimY.
+   * @todo is yDirMeansDown really necessary? It is only present in this file.
    *
    * @returns Nothing, acts in-place.
    */
   propagatePhotons(yDirMeansDown = true): void {
     const photonPropagator = this.createPhotonPropagator(yDirMeansDown)
     _.range(this.nPhotons).forEach(i => {
-      this.vector = photonPropagator.mulVecPartial(this.vectorPosDirIndicesForParticle(i), this.vector)
+      this.vector = photonPropagator.mulVecPartial(this.vectorPosDirIndicesForPhoton(i), this.vector)
     })
   }
 
   /**
    * Create an operator for a particular place, projecting only on the particular position.
-   * @param op Operator, assumed to be with dimensions [pol, dir].
+   * @param ILocalOperator a local operator interface containing:
    * @param posX Position x.
    * @param posY Posiiton y.
+   * @param op Operator, assumed to be with dimensions [pol, dir].
    *
    * @returns An operator [dimX, dimY, pol, dir].
    */
-  createLocalizedOperator(op: Operator, posX: number, posY: number): Operator {
-    return Operator.outer([Operator.indicator([this.dimX, this.dimY], [`${posX}`, `${posY}`]), op])
+  createLocalizedOperator(op: ILocalOperator): Operator {
```

Comment:
As before, not a big fan of wrapping all arguments.

(See, now the TSDoc is not compatible - it is incorrect to specify `posX` act as `@param`.

Reply:
It uses an interface as required in the todo. (Can rewrite the TS doc no problem)